src/pymor/operators/constructions.py

```diff
@@ -83,45 +83,81 @@ def evaluate_coefficients(self, mu):
         return [c.evaluate(mu) if hasattr(c, 'evaluate') else c for c in self.coefficients]
 
     def apply(self, U, mu=None):
-        coeffs = self.evaluate_coefficients(mu)
-        R = self.operators[0].apply(U, mu=mu)
-        R.scal(coeffs[0])
-        for op, c in zip(self.operators[1:], coeffs[1:]):
-            R.axpy(c, op.apply(U, mu=mu))
+        coeffs = np.array(self.evaluate_coefficients(mu))
+        if not coeffs.any():
+            R = self.range.zeros()
+        else:
+            where_nonzero = coeffs.nonzero()[0]
+            if where_nonzero[0]==0:
+                R = self.operators[0].apply(U, mu=mu)
+                R.scal(coeffs[0])
+            else:
+                R = self.range.zeros()
+            for idx in where_nonzero[max(0,1-where_nonzero[0]):]:
+                R.axpy(coeffs[idx], self.operators[idx].apply(U, mu=mu))
```

Comment:
This seems a bit complicated. What about
```python
coeffs = self.evaluate_coefficients(mu)
if coeffs[0]:
    R = self.operators[0].apply(U, mu=mu)
    R.scal(coeffs[0])
else:
    R = self.range.zeros()
for op, c in zip(self.operators[1:], coeffs[1:]):
    if c:
        R.axpy(c, op.apply(U, mu=mu))
```

Reply:
Yes you are right this is way simpler and was actually the first draft. 
With the 'complicated' code I wanted to avoid a python loop over all coeffs. Probably does not make much of a difference though.